### ISSUE TITLE
[CoE Starter Kit - BUG] Redo Portals flow with their new architecture

### ISSUE THREAD

**Issue by BenUnlox** [microsoft/coe-starter-kit#8038]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

Power Pages created In Developer environments aren't inventoried - other Environment types are not impacted by this. 

### Expected Behavior

Developer Environments to be fully inventoried. 

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.29

### What app or flow are you having the issue with?

?

### What method are you using to get inventory and telemetry?

None

AB#2869

**Comment by Jenefer-Monroe** [comment]
Please find a target envt where you expect to see power pages inventoried but do not. 
Then can you please check these two things for me? 

1. System Admin SR?
Are you listed with the System Admin SR in the envt? 

2. Envt Properties
Please find the environment in the CoE inventory table, and add the columns shown so we can understand current state for these properties:
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/f3146ed9-0f7f-4782-96e3-1424abfc3bc9)

**Comment by BenUnlox** [comment]
Thanks for getting back to me!

1.  Yes - the account has the SysAdmin SR assigned. 
2. ![Screenshot 2024-04-10 at 16 27 36](https://github.com/microsoft/coe-starter-kit/assets/108944628/b3cb15dc-2a57-4b02-8c8c-f47a7ca8ae9b) (Hidden the Environment Name but happy to provide privately if needed)

**Comment by Jenefer-Monroe** [comment]
Can you go find the flow, add displayname as a column to view, and find this envt? You will likely need to hit "All Runs" and "Show More"
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/7c89a349-dedf-43bf-831f-6d3ff671d470)

Once you find it click the timestamp to open the run and let me see what you see.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/717ac2df-aaab-43e1-9b6c-8afb42733a64)

**Comment by BenUnlox** [comment]
![Screenshot 2024-04-11 at 10 35 59](https://github.com/microsoft/coe-starter-kit/assets/108944628/d5d6e4ee-e16e-4a8b-85b2-35bb2d4e726a)
![Screenshot 2024-04-11 at 10 37 57](https://github.com/microsoft/coe-starter-kit/assets/108944628/3e651630-f48f-41dc-825d-bb1655c0b9e2)

Full Body:

> {
>   "error": {
>     "code": "0x80060888",
>     "message": "Resource not found for the segment 'adx_websites'."
>   }
> }

**Comment by Jenefer-Monroe** [comment]
That is the expected behavior if there are no portals in the envt. 

Can you please look to see if there are Websites listed for the envt? Here from a dev envt in our test tenant
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/745e0ed1-fe38-4427-98ec-13e52f4b0943)

And then if there are Power Pages there, look in the tables to see if this table is present:
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/2a76b144-810d-4f49-8c21-03cc332b3e84)

**Comment by BenUnlox** [comment]
In the example Environment given there are 3 Power Pages that do appear in both the locations mentioned in the previosu response:

![Screenshot 2024-04-11 at 13 59 54](https://github.com/microsoft/coe-starter-kit/assets/108944628/39680385-e405-4740-8ab8-2e508e03599a)
![Screenshot 2024-04-11 at 14 00 37](https://github.com/microsoft/coe-starter-kit/assets/108944628/4b0e88e7-b68c-41bb-8ed6-6db1e4a782fd)

**Comment by Jenefer-Monroe** [comment]
It seems like these are in this table instead yes?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/2dd45f4f-8b8a-4c48-a773-957709ee9fde)

**Comment by BenUnlox** [comment]
Ah apologies! The screenshot above were from the 'mspp_website' table, the 'adx_website' table doesn't exist in this Environment.

![Screenshot 2024-04-12 at 09 41 45](https://github.com/microsoft/coe-starter-kit/assets/108944628/ae6d1605-5f68-446e-9dc1-510600259b2e)

When I've checked other Environments, I couldn't find any that were created after 2021 that have the 'adx_website' table. In case this Table is added when a Power Page is created, I've created a Power Page in both a Developer and Production Environment - but in both this table still isn't present. The 'mspp_website' table is present in all Environments. Do we know if the 'adx_website' table get deprecated/replaced with the 'mspp_website' table?

**Comment by Jenefer-Monroe** [comment]
Yes it looks like they have moved forward with a deprecation that I was unaware of. 
We will have to redo our Portals work with this. Thanks for raising! Putting as a feature to do.

**Comment by miguelverweij** [comment]
I wanted to raise a new Feature Request, but I found this thread fortunately. The current Sync flow runs over Power Pages that use the adx prefix (adx_websites). this prefix is used for the "classic" Power Pages. If we want to put the Power Page Site in a solution, we need to enable the Enhanced Data Model. This is a different data model and stores the Power Pages Site in a tabel called mspp_website. The result is that all Enhanced Data Model Power Pages Site are currently out of scope for the inventory flow. @Jenefer-Monroe can you tell why this has the blocked status?

**Comment by Jenefer-Monroe** [comment]
It requires the new **_Power Platform for Admins V2_** connector and it is not yet rolled to all regions. 
There are other features in there related to the new PIM feature which have forced it to ship to everywhere soon. 
Fingers crossed on that.

**Comment by martincyr** [comment]
@J@Jenefer-Monroe any update on this now that the _Power Platform for Admins V2_ connector seems to be rolled out?

**Comment by Jenefer-Monroe** [comment]
sorry I no longer own the CoE STarter Kit, to Grant for consideration.

**Comment by Coding-Chem-Kitten** [comment]
Hi, @Grant-Archibald-MS , Are there any updates on this? Thanks!

**Comment by Grant-Archibald-MS** [comment]
@Coding-Chem-Kitten I am containing to follow up on this with the owners of this connector.

One approach we are looking at is creating a minimalist sample that just uses the Power Platform for Admins V2 connector to verify its setup across different environments

**Comment by Coding-Chem-Kitten** [comment]
@Grant-Archibald-MS Thanks for your reply.  Would it be feasible to update the flow to recognize both adx_ and mspp_ table prefixes? As a temporary solution, I duplicated the Admin | Sync Template v4 Portals flow and replaced all instances of the adx_ prefix with mspp_, and it seems to be working for us for now.

Would a change like that be a possible fix for you to include in the next CoE update?

**Comment by RajeevPentyala** [comment]
@Coding-Chem-Kitten We are already updating the flow to handle both the Standard (Adx_) and Enhanced Data Model (mspp_). We’ll try to include it in the May release, provided it passes testing. Thanks.

**Comment by AmarSaiRam** [comment]
Please proceed with upgrading to the latest version of the Center of Excellence (CoE) Starter Kit. Before initiating the update, make sure to identify and remove all unmanaged layers to prevent deployment issues.
The necessary fixes have already been applied, so installing the most recent version of the CoE Starter Kit is likely to resolve the problems you're facing and improve functionality

<img width="448" height="255" alt="Image" src="https://github.com/user-attachments/assets/f217badc-c177-4f6d-87f3-368180409d73" />
